air/src/execution_step/boxed_value/stream_map.rs

```diff
@@ -83,7 +83,30 @@ impl StreamMap {
     }
 
     pub(crate) fn get_unique_map_keys_stream(&mut self) -> Cow<'_, Stream> {
-        self.stream.get_unique_map_keys_stream()
+        use std::collections::HashSet;
+
+        let mut distinct_keys = HashSet::new();
+
+        // unwrap is safe because slice_iter always returns Some iff generation slice is valid
+        let new_values = self
+            .stream
+            .slice_iter(Generation::Nth(0.into()), Generation::Last)
+            .unwrap()
+            .map(|values| {
+                values
+                    .iter()
+                    .filter(|v| {
+                        StreamMapKey::from_kvpair(v)
+                            .map(|key| distinct_keys.insert(key))
+                            .unwrap_or(false)
+                    })
+                    .cloned()
+                    .collect::<Vec<_>>()
+            })
+            .collect::<Vec<_>>();
+
+        let stream = Stream::new(new_values, self.stream.previous_gens_count());
```

Comment:
It seems that explicit types are not needed in both `.collect` above, as the type is deduced from `Stream::new` type, buy may be kept for documentation proposes.

Reply:
Yep, it's mostly for documentation purposes.